### PR TITLE
chore: update sdk version

### PR DESCRIPTION
**Pull request by bogdan-manole**
update sdk version from beta to stable

### CODE REVIEW

**Comment by marc0olo** [comment]
@bogdan-manole can we directly move to v11.0.1 please?

**Comment by bogdan-manole** [comment]
> @bogdan-manole can we directly move to v11.0.1 please?

With v11.0.1 I have some errors when I run test  regarding events decoding. I will dig later to see what's happening there

**Comment by marc0olo** [comment]
@bogdan-manole any plans to revisit this in the near future? we should directly move to v12.0.0 then 😅

**Comment by bogdan-manole** [comment]
> @bogdan-manole any plans to revisit this in the near future? we should directly move to v12.0.0 then 😅

Yup, I will try to update it these days.

**Comment by bogdan-manole** [comment]
- there are two commits, the first for migrating to `v12`  and the second to `v13` ( we can squash the commits, but I thought it could be used if someone wants to use the v12 at some point in time )
- in the previous versions there was a `decodeEvents` function attached to the `Contract` instance. 
  It was helpful because one could decode some unknown events (from a contract remote call) with the right contract object.
  For example, you have an instance of the `AedexRouter` contract, in the call of `add_liquidity` of this contract, a `Transfer` event was produced of an `AEX9` token. You could use the `decodeEvents` belonging to an `AEX9` instance instead of the `AedexRouter`. 
Now (in v13) there is now `decodeEvents` to decode in the first place with other contract instances. But also, now there was an error when for a contract call which didn't have the definition of the `event`. Unless `omitUnknown` was set to true, that meant we could not test most of the events (the `router` for example doesn't have its own events, all that happens is inside `AedexPair` and the `tokens`)
The solution was to add those events into the definition of the interfaces used by the parent contract. For example, there has been added the `AEX9` events into the `IAEX9Minimal`. 
**The question is**: even if no functional codebase was changed in the contracts, is the audit impacted by that?

**Comment by marc0olo** [comment]
> * **The question is**: even if no functional codebase was changed in the contracts, is the audit impacted by that?

IMO no, I mean the audit is targeting the (functional) contract codebase, right?

I think a changelog to contract after audit with respective commit id would be good to provide full transparency in that regards

**Comment by kenodressel** [comment]
If we use the full ACI, will it not contain all the remote events? Otherwise I would consider this a bug. The SDK should not fail just because it can not decode the events.

File: contracts/interfaces/IAEX9Minimal.aes [hunk 1]
```diff
@@ -7,6 +7,16 @@ contract interface IAEX9Minimal =
 
     record meta_info = { name : string , symbol : string , decimals : int }
 
+    datatype event
```
Comment (thepiwo) [inline contracts/interfaces/IAEX9Minimal.aes:10]: I'd just add this in the ACI, not in the interface itself

File: contracts/interfaces/IAedexV2Pair.aes [hunk 2]
```diff
@@ -8,6 +8,24 @@ include "./IAedexV2FactoryForPair.aes"
 contract interface IAedexV2Pair =
     entrypoint balance: (address) => option(int)
 
+    datatype event
```
Comment (thepiwo) [inline contracts/interfaces/IAedexV2Pair.aes:11]: same here, lets just add it in the aci

File: contracts/AedexV2Pair.aes [hunk 3]
```diff
@@ -1,5 +1,3 @@
-@compiler >= 6
```
Comment (thepiwo) [inline contracts/AedexV2Pair.aes:1]: no need to remove

File: deployment/aci/AedexV2Factory.aci.js [hunk 4]
```diff
@@ -0,0 +1,785 @@
+ /* eslint-disable */
+module.exports = [
```
Comment (thepiwo) [inline deployment/aci/AedexV2Factory.aci.js:2]: why js over json?

File: deployment/deploy.js [hunk 5]
```diff
@@ -14,66 +14,70 @@
...
+
+    const cmp = new CompilerHttp( compiler ? compiler : NETWORKS[NETWORK_NAME].compilerUrl )
+    const client = new AeSdk( {
+        nodes      : [ { name: NETWORK_NAME, instance } ],
+        onCompiler : cmp,
+        interval   : 50,
     } )
+
+    const accounts = [ new MemoryAccount(  secretKey )  ]
+    await Promise.all(
```
Comment (thepiwo) [inline deployment/deploy.js:42]: this is not needed any more

File: deployment/deploy.js [hunk 6]
```diff
@@ -14,66 +14,70 @@
...
-                } )
-            },
-        ],
-        compilerUrl : compiler ? compiler : NETWORKS[NETWORK_NAME].compilerUrl,
-        accounts    : [ MemoryAccount( { keypair: KEYPAIR } ) ],
-        address     : KEYPAIR.publicKey
+    const instance = new Node( NETWORKS[NETWORK_NAME].nodeUrl, { ignoreVersion: true } )
+
+    const cmp = new CompilerHttp( compiler ? compiler : NETWORKS[NETWORK_NAME].compilerUrl )
+    const client = new AeSdk( {
```
Comment (thepiwo) [inline deployment/deploy.js:35]: or you use aeproject utils.getSdk()

File: contracts/interfaces/IAEX9Minimal.aes (hunk 1 above)
Comment (bogdan-manole) [reply contracts/interfaces/IAEX9Minimal.aes:10]: In tests at least, we are compiling the router with all the other dependent contracts (DexFactory, DexPairs, and interfaces, including IAEX9Minimal.aes), after the compilation we deploy the contract, and as a result, we get the contract instance we are using it for tests. Is there a place for injecting some extra ACI, in this process? 
Also, those events are part of  IAEX9 standard

File: deployment/deploy.js (hunk 5 above)
Comment (bogdan-manole) [reply deployment/deploy.js:42]: done

File: contracts/AedexV2Pair.aes (hunk 3 above)
Comment (bogdan-manole) [reply contracts/AedexV2Pair.aes:1]: done

File: deployment/aci/AedexV2Factory.aci.js (hunk 4 above)
Comment (bogdan-manole) [reply deployment/aci/AedexV2Factory.aci.js]: done (changed to json format)

File: deployment/deploy.js (hunk 6 above)
Comment (bogdan-manole) [reply deployment/deploy.js:35]: covered by future issue #83

File: docker-compose.yml [hunk 7]
```diff
@@ -12,7 +12,7 @@ services:
       - './accounts_test.json:/home/aeternity/node/data/aecore/.genesis/accounts_test.json'
   compiler:
     hostname: compiler
-    image: 'aeternity/aesophia_http:v6.1.0'
```
[the remark this answers is not in the document]
Comment (bogdan-manole) [reply docker-compose.yml:15]: covered by issue #83

File: test/shared/fixtures.js [hunk 8]
```diff
@@ -359,8 +387,8 @@ const pairFixture = async ( wallet = wallet0 ) => {
     return ret
 }
 const awaitOneKeyBlock = async ( client ) => {
```
[the remark this answers is not in the document]
Comment (bogdan-manole) [reply test/shared/fixtures.js:389]: covered by issue #83

File: test/shared/fixtures.js [hunk 9]
```diff
@@ -372,20 +400,17 @@ const beforeEachWithSnapshot = ( str, work ) => {
         await work()
         console.debug( "initial work ... DONE " )
         // get the snapshot height
-        //snapshotHeight = await getBlockHeight()
-        snapshotHeight = await client.height()
+        snapshotHeight = await client.getHeight()
```
[the remark this answers is not in the document]
Comment (bogdan-manole) [reply test/shared/fixtures.js:403]: covered by issue #83

**Review by thepiwo — APPROVED** [review]

**Pull request merged**